cmd/skaffold/app/cmd/fix.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/GoogleCloudPlatform/skaffold/pkg/skaffold/config"
+	"github.com/GoogleCloudPlatform/skaffold/pkg/skaffold/schema"
+	"github.com/GoogleCloudPlatform/skaffold/pkg/skaffold/util"
+)
+
+func NewCmdFix(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "fix",
+		Short: "Converts old skaffold.yaml to newest schema version",
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg, err := util.ReadConfiguration(filename)
+			if err != nil {
+				logrus.Errorf("fix: %s", err)
+			}
+			version, err := getApiVersion(cfg)
+			if err != nil {
+				logrus.Errorf("fix: %s", err)
+			}
+			if version == config.LatestVersion {
+				fmt.Println("config is already latest version")
+				return
+			}
+			if err := runFix(cfg); err != nil {
+				logrus.Errorf("fix: %s", err)
+			}
+		},
+		Args: cobra.NoArgs,
+	}
+	return cmd
+}
```

Comment:
lets add some flags to this command

string
--f, --filename, the path of the config file to fix, defaults to `skaffold.yaml`

--overwrite=true/false, whether or not to overwrite the config with the fixed one, default to false.

Reply:
alright, if we don't overwrite we have two options: write to a new file (`<filename>.new`), or print to stdout but don't write a file. I personally like writing to a new file, WDYT?